nestkernel/archiving_node.h

```diff
@@ -93,6 +93,7 @@ class ArchivingNode : public StructuralPlasticityNode
    * with t > t_first_read.
    */
   void register_stdp_connection( double t_first_read, double delay ) override;
+  void register_eprop_connection( double t_first_read, double delay ) override;
```

Comment:
Since `register_stdp_connection` has "stdp" in the name, it might be misleading to use it for e-prop plasticity which does not have spike-dependence. Nonetheless, you're right that the functions do the same. A solution would be rename `register_stdp_conenction` to `register_connection`, use it for e-prop and get rid of `register_eprop_connection`.

Reply:
The name has historic roots based on where it was first used, not what it does or means. I would suggest to rename `register_stdp_connection()` to `register_spike_history_client()` (or `..._reader`, `..._subscriber`) to express what the call is really about. That will fit equally well for plastic and eprop connections.